benches/benches/bevy_ecs/observers/propagation.rs

```diff
@@ -62,14 +62,16 @@ pub fn event_propagation(criterion: &mut Criterion) {
 }
 
 #[derive(EntityEvent, Clone, Component)]
-#[entity_event(traversal = &'static ChildOf, auto_propagate)]
-struct TestEvent<const N: usize> {}
+#[entity_event(propagate, auto_propagate)]
+struct TestEvent<const N: usize> {
+    entity: Entity,
```

Comment:
This is going to be a *painful* breaking change, and increase boilerplate.

I do appreciate that it allows for better docs, and is a lot less magical, which is good.